### ISSUE TITLE
Improve documentation of the `showTooltip` prop in various components

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#66523]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/pull/60908#discussion_r1814809336

It would be good to add some guidance about its effective usage, specifically in combination with the case when the 'Show button text labels' preferences (`showIconLabels`) is enabled. It's worth considering explaining the a11y best practices that surround this and help to guide future contributors.

It's wirth reminding that when `showIconLabels` is enabled, icon buttons switch to show visible text and in this case showing a tooltip that just repeats the visible text is unnecessary clutter. Example:

![Screenshot 2024-04-19 at 15 22 29](https://github.com/WordPress/gutenberg/assets/1682452/9fd8208f-ea50-4284-a748-3b4f797df4aa)

Cc @getdave 

### Step-by-step reproduction instructions

N/A

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by afercia** [comment]
Occurrences found in the documentation:

```
packages/components/src/button/README.md:
  252: #### `showTooltip`: `boolean`

packages/components/src/range-control/README.md:
  324: ### `showTooltip`: `boolean`

packages/components/src/toggle-group-control/toggle-group-control-option/README.md:
  51: ### `showTooltip`: `boolean`

packages/components/src/toggle-group-control/toggle-group-control-option-base/README.md:
  23: ### `showTooltip`: `boolean`
```

**Comment by Vrishabhsk** [comment]
Hi @afercia 👋 
- I have raised a `PR` to improve the docs for `Button`, `Toggle Group Control (option & option-base)` : [#66673](https://github.com/WordPress/gutenberg/pull/66673)

### Doubts
- When I implemented the `Button` component and enabled the `showIconLabels` preference, the button did not change
- After some evaluation, found that the styles responsible for this behavior are only limited to a few of the editor selectors. Is this expected?
- Also for `rangeControl`, the prop is for either forcing the tooltip to show/hide which is out-of-scope in relation to the `showIconLabels` preference, so I have ignored it for now. Let me know if I'm mistaken here. 

Let me know your thoughts on this. Thanks

**Comment by afercia** [comment]
> After some evaluation, found that the styles responsible for this behavior are only limited to a few of the editor selectors. Is this expected?

Yes, unfortunately the `showIconLabels` implementation is partial, so far. A while ago I created https://github.com/WordPress/gutenberg/issues/61763 to explore some more solid implementation but there hasn't been much movement so far.

> Also for rangeControl, the prop is for either forcing the tooltip to show/hide

Yes right, in this case the tooltip is useed to show the value while dragging the range 'thub'. Unrelated.
